beard/similarity/pairs.py

```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Beard.
+# Copyright (C) 2015 CERN.
+#
+# Beard is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+"""Transformers for paired data.
+
+.. codeauthor:: Gilles Louppe <g.louppe@cern.ch>
+
+"""
+import numpy as np
+import scipy.sparse as sp
+
+from sklearn.base import TransformerMixin
+
+
+class PairTransformer(TransformerMixin):
+
+    """Apply a transformer on all elements in paired data."""
+
+    def __init__(self, element_transformer):
+        """Initialize.
+
+        Parameters
+        ----------
+        :param element_transformer: transformer
+            The transformer to apply on each element.
+        """
+        self.element_transformer = element_transformer
+
+    def fit(self, X, y=None):
+        """Fit the given transformer on all individual elements in `X`.
+
+        Rows i in `X` are assumed to represent pairs, where
```

Comment:
I think doubled backquotes should be used for code samples. It is worth checking whole Beard code for such misusages.

http://sphinx-doc.org/rest.html